runner/src/runner.rs

```diff
@@ -1,45 +1,85 @@
-use std::process::Command;
+use std::process::{Command, Stdio};
 use std::path::PathBuf;
 use std::cell::RefCell;
 use std::rc::Rc;
 use std::fs::File;
 use std::io::Read;
 use std::fs::remove_file;
 use std::collections::HashMap;
+use std::time::{Duration, Instant};
+use wait_timeout::ChildExt;
 
 /// Runner allows to execute the testsuite with the target mutation count
 pub trait Runner {
     /// run executes the testsuite with the given mutation count and returns the output
     /// if all tests has passed
-    fn run(&self, mutation_count: usize) -> Result<(), ()>;
+    fn run(&mut self, mutation_count: usize) -> Result<(), ()>;
+}
+
+pub struct RuntimeModifier {
+    multiplier: u32,
+    divisor: u32,
+    baseline: Duration,
+}
+
+impl RuntimeModifier {
+    fn new(mul: u32, div: u32, baseline: Duration) -> RuntimeModifier {
+        RuntimeModifier {
+            multiplier: mul,
+            divisor: div,
+            baseline
+        }
+    }
+
+    fn compute(&self, duration: Duration) -> Duration {
+        duration * self.multiplier / self.divisor + self.baseline
+    }
 }
 
 /// Full suite runner executes all the test at once, given the path of the executable
 pub struct FullSuiteRunner {
     test_executable: PathBuf,
+    runtime_mod: RuntimeModifier,
+    timeout: Duration,
 }
 
 impl FullSuiteRunner {
     /// creates a runner from the test executable path
     pub fn new(test_executable: PathBuf) -> FullSuiteRunner {
         FullSuiteRunner {
-            test_executable
+            test_executable,
+            runtime_mod: RuntimeModifier::new(5, 2, Duration::from_millis(250)),
+            timeout: Duration::from_secs(0),
         }
     }
 }
 
 impl Runner for FullSuiteRunner {
-    fn run(&self, mutation_count: usize) -> Result<(), ()> {
-        let output = Command::new(&self.test_executable)
-            // 0 is actually no mutations so we need i + 1 here
-            .env("MUTATION_COUNT", mutation_count.to_string())
-            .output()
-            .expect("failed to execute process");
-
-        if output.status.success() {
-            Ok(())
+    fn run(&mut self, mutation_count: usize) -> Result<(), ()> {
+        let mut command = Command::new(&self.test_executable);
+        command.env("MUTATION_COUNT", mutation_count.to_string())
+            .stdout(Stdio::null());
+        if mutation_count == 0 {
+            let start = Instant::now();
+            let status = command.status().expect("failed to execute process");
```

Comment:
Alas, it cannot, because else we'd return an `Err(_)` (which means the test failed) where we should really stop the world. Perhaps we should change it to return `Result<bool, Error>`?